jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java

```diff
@@ -509,13 +513,59 @@ protected void copyRequestHeaders(HttpServletRequest clientRequest, Request prox
 
     protected void addProxyHeaders(HttpServletRequest clientRequest, Request proxyRequest)
     {
-        addViaHeader(proxyRequest);
+        addViaHeader(clientRequest, proxyRequest);
         addXForwardedHeaders(clientRequest, proxyRequest);
     }
 
+    /**
+     * Adds the HTTP Via header to the proxied request.
+     *
+     * @deprecated Use {@link #addViaHeader(HttpServletRequest, Request)} instead.
+     * @param proxyRequest the request being proxied
+     */
+    @Deprecated
     protected void addViaHeader(Request proxyRequest)
     {
-        proxyRequest.header(HttpHeader.VIA, "http/1.1 " + getViaHost());
+        oldAddViaHeaderCalled = true;
```

Comment:
This is not good enough because the overridden method might just be:
```java
    protected void addViaHeader(Request proxyRequest)
    {
        System.err.println("addViaHeader called: " + proxyRequest);
        super.addViaHeader(proxyRequest);
    }
```
So the method still needs to work!